File: libsignal/service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/BlockedListMessage.java
package org.whispersystems.signalservice.api.messages.multidevice;

import org.whispersystems.signalservice.api.push.SignalServiceAddress;

import java.util.List;

public class BlockedListMessage {

  private final List<SignalServiceAddress> addresses;
  private final List<byte[]>               groupIds;

  public BlockedListMessage(List<SignalServiceAddress> addresses, List<byte[]> groupIds) {
    this.addresses = addresses;
    this.groupIds  = groupIds;
  }

  public List<SignalServiceAddress> getAddresses() {
    return addresses;
  }

  public List<byte[]> getGroupIds() {
    return groupIds;
  }
}

File: lib/libsignal-service/src/main/java/org/whispersystems/signalservice/api/messages/multidevice/BlockedListMessage.kt
package org.whispersystems.signalservice.api.messages.multidevice

import org.signal.core.models.ServiceId

class BlockedListMessage(
  @JvmField val individuals: List<Individual>,
  @JvmField val groups: List<Group>
) {
  data class Individual(
    val aci: ServiceId.ACI?,
    val e164: String?,
    val blockedAt: Long
  ) {
    init {
      check(aci != null || e164 != null)
    }
  }

  data class Group(
    val groupId: ByteArray,
    val blockedAt: Long
  )
}
